app/Handlers/Utility/CurrencyHandler.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+namespace App\Handlers\Utility;
+
+use App\Slack\Event;
+use App\Slack\Message;
+use App\Handlers\Handler;
+use App\Loader\LoadsData;
+use App\Loader\JsonLoader;
+use Symfony\Component\Intl\Intl;
+use Illuminate\Support\Collection;
+
+final class CurrencyHandler extends Handler
+{
+    use LoadsData;
+
+    protected $dataFile = 'currency.json';
+
+    /**
+     * @param JsonLoader $loader
+     */
+    public function __construct(JsonLoader $loader)
+    {
+        $this->loader = $loader;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function canHandle(Event $event)
+    {
+        return
+            $event->isMessage() &&
+            ($event->isDirectMessage() || $event->mentions($this->eve->userId())) &&
+            $event->matches('/convert .+/i')
+        ;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(Event $event)
+    {
+        $this->loadData();
+
+        $arguments = $this->arguments($event);
+        $content   = '`@eve convert 10 GPB to USD`';
+        $baseRate  = 1;
+
+        if ($arguments->count() == 4 && $this->validArguments($arguments)) {
+            $rates  = $this->data['rates'];
+            $result = '';
+
+            if (! array_key_exists($arguments[1], $rates) && ! array_key_exists($arguments[3], $rates)) {
```

Comment:
Rather than using the array indices for these checks, extract the data into variables to make this more readable. For example:

```
if (! array_key_exists($baseCurrency, $rates...
```

That's way easier to read and understand what's going on than the above.